Cloud logs page (#67)
loadimpact/k6-docs: src/data/markdown/docs/03 cloud/02 Analyzing Results/11 Cloud Logs.md

---
title: 'Cloud Logs'
excerpt: 'Execution logs and Troubleshooting'
---

## Background

The execution logs functionality is currently in Beta (not guaranteed to be stable). Execution logs will be available natively in k6 0.28.0 (planned release September 2020).

The logs will also be viewable directly in the frontend app in the "Execution logs" tab (planned release: August 15). Meanwhile, logs can be viewed in real-time using the "k6cloudlogs" command-line utility.

The execution logs are intended to aid you in troubleshooting your script. The logs are deleted 3 days after the test execution.

## Installation

> k6cloudlogs binaries for Linux, Windows and MacOS can be found here: https://github.com/loadimpact/k6cloudlogs/releases/

Reviewer: ```suggestion
k6cloudlogs binaries for Linux, Windows and macOS can be found here: https://github.com/loadimpact/k6cloudlogs/releases/
```

## Using k6cloudlogs CLI utility

```bash
Usage of ./k6cloudlogs:
  -id string
        test run id
  -level string
        lowest logging level to return events for (info assumes error, etc.) (default "info")
  -limit string
        maximum amount of messages to be in a response from the server (default "100")
  -start string
        from how long ago to start tailing (default "5m")
  -token string
        k6 Cloud authentication token
```

> Setting the token can also be done from the env variable K6_CLOUD_TOKEN

Reviewer: ```suggestion
Setting the token can also be done from the environment variable `K6_CLOUD_TOKEN`.
```

## Basic example

> To tail the execution logs from your cloud test in real-time you have to provide a testrunid and your API token like this:

Reviewer: ```suggestion
To tail the execution logs from your cloud test in real-time you have to provide a test run ID and your API token like this:
```

```bash

> ./kcloudlogs -id 127855 -token YOUR_API_TOKEN

Reviewer: Let's showcase the safer one ;)

```suggestion
K6_CLOUD_TOKEN=<YOUR_API_TOKEN> ./kcloudlogs -id 127855
```

```

You can find your API token in https://app.k6.io/account/api-token. 

The output should look similar to this:

![Example output](/images/11%PCloug%Logs/cloud-logs-example-output.png)

## Fetching logs from the past

By default, the k6cloudlogs utility fetches logs in real-time. This means that you need to have it running when your test is executing.
If you would like to retrieve logs from the past, you need to provide a `-start` parameter. 

For example:

```bash
K6_CLOUD_TOKEN=YOUR_API_TOKEN./kcloudlogs -id 127855 -start 55m -limit 1000
```

This command will retrieve 1000 log lines from 55 minutes ago. Note, only 1000 lines will be printed. The limit for the `-limit` parameter is about 1500. 

> If you would like to view more logs, you will have to execute this command several times with different `-start` parameter. 

Reviewer: ```suggestion
If you would like to view more logs, you will have to execute this command several times with a different `-start` parameter.
```
